### DISCUSSION TITLE
DatePicker issues while migrating from PrimeFaces v8.0 up to v.12.0

### DISCUSSION THREAD

**Discussion by ViniciusLagoGehrke** [primefaces/community#821] [PrimeFaces] [answered]
I was assigned a task to fix the whole UI after _PrimeFaces_ and it's theme _Morpheus_ had their versions updated. Most of it I was able to solve just overwriting the CSS and replacing deprecated components for new ones.
The 3 main issues are with the _Calendar_ component, now _DatePicker_:

- The before and after arrows at the header are gone (although I can click on it and change the month)
- The time picker does not allow edition, neither from the _ui-picker-up_ or _ui-picker-down_ nor the _span_ number itself
- The up and down time pickers doesn't appear either

Here you can see how it was before the version update:
<img width="639" alt="image" src="https://github.com/primefaces/community/assets/46430040/8ccf3e74-8f5f-4aa2-8f97-b6e0bb062b98">

and after (failling):
<img width="645" alt="image" src="https://github.com/primefaces/community/assets/46430040/ff8535a8-68f2-4fff-9a6b-dfe75ebc169c">

Here is the datepicker at the xhtml:
```
<p:datePicker
    id="fromCalendar"
    value="#{entranceListBean.from}"
    pattern="#{localeBean.getDateTimePattern()}"
    lang="#{localeBean.locale.language}"
    readonlyInput="true"
    showTime="true"
  />
    <p:ajax global="false" ignoreAutoUpdate="true"/>
  </p:datePicker>
-
  <p:datePicker
    value="#{entranceListBean.to}"
    pattern="#{localeBean.getDateTimePattern()}"
    lang="#{localeBean.locale.language}"
    showTime="true"
  >
    <p:ajax global="false" ignoreAutoUpdate="true"/>
  </p:datePicker>
```

And here is constructor of the _EntranceListBean.java_:
```
public void init() {
        Calendar calendar = GregorianCalendar.getInstance();
        calendar.setTimeInMillis(System.currentTimeMillis());
        calendar.set(Calendar.HOUR_OF_DAY, 0);
        calendar.set(Calendar.MINUTE, 0);
        calendar.set(Calendar.SECOND, 0);
        calendar.set(Calendar.MILLISECOND, 0);
        from = new Date(calendar.getTimeInMillis());
        calendar.add(Calendar.DATE, 1);
        calendar.add(Calendar.MILLISECOND, -1);
        to = new Date(calendar.getTimeInMillis());
}
```

**Comment by melloware** [comment]
Why go to 12.0.0 when 13.0.4 is out and probably these bugs are already fixed?

**Reply by melloware** [reply]
Oh your screenshot is still using `Calendar` not `DatePicker`?

**Comment by ViniciusLagoGehrke** [comment]
> Oh your screenshot is still using `Calendar` not `DatePicker`?

Where am I using Calendar? The only place that I find it is inside the .java file which actually refers to _Java.utils.Calendar_

**Reply by melloware** [reply]
Oh sorry was looking at your first screenshot.  Yeah in your second screenshot the issue is your theme. The dark blue of the theme is hiding the previous and next buttons on the DatePicker.  Your Morpheus theme needs DatePicker CSS tweaks.

**Reply by melloware** [reply]
Are you sure you downloaded the latest Morpheus Theme from PrimeStore that matches PF 12 or 13?

**Reply by ViniciusLagoGehrke** [reply]
Investigating the Morpheus Theme now. Get back here with the results.

**Reply by ViniciusLagoGehrke** [reply, marked as answer]
Morpheus version is now 5.1.0 (previously 1.0.4)
We're not able to use PrimeFaces 13.0.0 that's why we're not using the last Morpheus version as well.

**Reply by melloware** [reply]
Got it but that explains it because in those Morpheus themes all the DatePicker CSS has been added since your 1.0.4.
